project_timesheet_holidays_contract/models/hr_employee.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from datetime import datetime
+
+from odoo import _, fields, models
+from odoo.exceptions import ValidationError
+
+
+class HrEmployee(models.Model):
+    _inherit = "hr.employee"
+
+    def _get_datetime_date(self, date):
+        return date if not isinstance(date, datetime) else date.date()
```

Comment:
I think it's unnecessary to have a new method just to convert `datetime` objects to `date`.

But if we really want to extract it, then I'd rather go with one of the following options:
1. use `@staticmethod` decorator
2. move this method outside of the class
3. move this method inside of `_get_calendar()` method